src/app/credExplorer/WeightConfig.js

```diff
@@ -55,43 +46,55 @@ export class WeightConfig extends React.Component<Props, State> {
               justifyContent: "space-between",
             }}
           >
-            {this.pluginWeightConfigs()}
+            {this._renderPluginWeightConfigs()}
           </div>
         )}
       </React.Fragment>
     );
   }
 
-  pluginWeightConfigs() {
+  _renderPluginWeightConfigs() {
     return this.props.adapters
       .adapters()
       .filter((x) => x.name() !== FALLBACK_NAME)
       .map((adapter) => {
         const onChange = (weightedTypes) => {
-          this.state.pluginNameToWeights.set(adapter.name(), weightedTypes);
-          this.fire();
+          const newWeightedTypes = {
```

Comment:
To clarify: this is like

```js
          const newWeightedTypes = {
            nodes: MapUtil.merge([
              MapUtil.copy(this.props.weightedTypes.nodes),
              weightedTypes.nodes,
            ]),
            edges: MapUtil.merge([
              MapUtil.copy(this.props.weightedTypes.edges),
              weightedTypes.edges,
            ]),
          };
```

except that you need to _replace_ duplicates instead of failing if they
exist, yes?

Reply:
That's right.